### PR TITLE
Apr 1848 improve updateable page process

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/GetApplicationByUserHandlerTests/When_application_by_user_is_requested.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApplyService.Application.Apply;
+using SFA.DAS.ApplyService.Application.Apply.GetApplications;
+
+namespace SFA.DAS.ApplyService.Application.UnitTests.Handlers.GetApplicationByUserHandlerTests
+{
+    public class When_applications_created_by_organisWhen_application_by_user_is_requested
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/GetApplicationByUserHandlerTests/When_application_by_user_is_requested.cs:11]: Typo in class name

File: src/SFA.DAS.ApplyService.Application/Apply/GetApplications/GetApplicationByUserHandler.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.ApplyService.Domain.Entities;
+
+namespace SFA.DAS.ApplyService.Application.Apply.GetApplications
+{
+    public class GetApplicationByUserHandler : IRequestHandler<GetApplicationByUserRequest, Domain.Entities.Apply>
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Application/Apply/GetApplications/GetApplicationByUserHandler.cs:11]: This is a bit petty could you name it: GetApplicationByUserIdHandler

File: src/SFA.DAS.ApplyService.Application/Apply/GetApplications/GetApplicationByUserRequest.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+using System;
+using MediatR;
+
+namespace SFA.DAS.ApplyService.Application.Apply.GetApplications
+{
+    public class GetApplicationByUserRequest : IRequest<Domain.Entities.Apply>
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Application/Apply/GetApplications/GetApplicationByUserRequest.cs:6]: This is a bit petty could you name it: GetApplicationByUserIdRequest

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 4]
```diff
@@ -59,6 +59,16 @@ public async Task<Apply> GetApplication(Guid applicationId)
             }
         }
 
+        public async Task<Apply> GetApplicationByUser(Guid applicationId, Guid signinId)
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:62]: This is a bit petty could you name it: GetApplicationByUserId

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/ApplicationController.cs [hunk 5]
```diff
@@ -67,6 +67,12 @@ public async Task<ActionResult<List<Domain.Entities.Apply>>> GetOrganisationAppl
             return await _mediator.Send(new GetApplicationsRequest(Guid.Parse(signinId), false));
         }
 
+        [HttpGet("Application/{applicationId}/Contact/{signinId}")]
+        public async Task<ActionResult<Domain.Entities.Apply>> GetOrganisationApplicationForUser(Guid applicationId, string signinId)
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/ApplicationController.cs:71]: The routing looks a bit off but I can't think of a better one.
Could we change the method name to: GetApplicationByUserId

File: src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs [hunk 6]
```diff
@@ -360,9 +360,7 @@ public void Confirm_submit_application_updates_application_status_and_sends_conf
                 }
             };
 
-            _apiClient.Setup(x => x.GetOrganisationByUserId(It.IsAny<Guid>())).ReturnsAsync(organisationDetails);
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs:363]: Is this supposed to be removed? I couldn't find the corresponding line removed in RoatpApplicationController

**Review by martin-hornsby — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs (hunk 6 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs:363]: No I mistook it for a call buried in the GetApplications handler, I'll reinstate

**Review by martin-hornsby — DISMISSED** [review]

**Review by martin-hornsby — APPROVED** [review]

**Pull request merged**